File: HackerRank-Data-Structures/Arrays & Linked Lists/Inserting a Node Into a Sorted Doubly Linked List.cpp
DoublyLinkedListNode* sortedInsert(DoublyLinkedListNode* head, int data) {
    if(head == nullptr)
        return new DoublyLinkedListNode(data);
    if(data <= head->data){
        DoublyLinkedListNode* newNode = new DoublyLinkedListNode(data);
        newNode->next = head;
        head->prev = newNode;
        return newNode;
    }
    DoublyLinkedListNode* newNode = sortedInsert(head->next, data);
    head->next = newNode;
    newNode->prev = head;
    return head;
}

File: HackerRank-Data-Structures/Arrays & Linked Lists/Inserting a Node Into a Sorted Doubly Linked List.py
def sortedInsert(head, data):
    if head == None:
        return DoublyLinkedListNode(data)
    if data <= head.data:
        newNode = DoublyLinkedListNode(data)
        newNode.next = head
        head.prev = newNode
        return newNode
    
    newNode = sortedInsert(head.next, data)
    head.next = newNode
    newNode.prev = head
    return head